compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp

```diff
@@ -967,6 +967,23 @@ IREE::GPU::TargetAttr getCLGPUTarget(MLIRContext *context) {
   return IREE::GPU::getFullTarget(backend, arch, features, context);
 }
 
+IREE::GPU::TargetAttr getGPUTargetAttr(Attribute attr) {
+  if (!attr) {
+    return {};
+  }
+  if (auto targetAttr = dyn_cast<IREE::HAL::ExecutableTargetAttr>(attr)) {
+    return getGPUTargetAttr(targetAttr);
+  } else if (auto dictAttr = dyn_cast<DictionaryAttr>(attr)) {
+    std::optional<NamedAttribute> targetAttr =
+        dictAttr.getNamed("iree.gpu.target");
+    if (!targetAttr) {
+      return {};
+    }
+    return cast<IREE::GPU::TargetAttr>(targetAttr->getValue());
+  }
+  return {};
```

Comment:
few style nits:

1. Don’t use else after a return. https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return
2. Prefer early exit to simplify code and save levels of nesting. https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code

so it can be:

```cpp
if (xxx) {
  return {};
}
if (xxx) {
  // ...
  return {}
}
return cast<...>(...);
```

Reply:
Done